jobs/build/oc_sync/publish-clients-from-payload.sh

```diff
@@ -1,12 +1,22 @@
 #!/bin/bash
```

Comment:
As the `promote` job copies quite some of the logic in this file, would it make sense to move it to `pipeline-scripts`, and have the `promote` job invoke that as well?

Reply:
i seem to recall the contexts were _just different enough_ to make it not worthwhile (we barely use `oc_sync` at all)

Reply:
I hoped we can use a consolidated logic to do this, but the promote job has its own oc_sync implementation rather than calling the oc_sync job. Don't really know why we have different implementation so I won't think it is a good idea to refactor in this PR.